### ISSUE TITLE
Flat config mode not detected when config is outside of root

### ISSUE THREAD

**Issue by GaborTorma** [nuxt/eslint#338]
I tried a new 0.3.0.alpha.0 and alpha.1 version in WSL Debian and checker: true. 
I'v got the following error if I try to run it in VS Code JS Debugger terminal. In bash, we have no error.

```
[3:42:59 PM]  ERROR  [uncaughtException] Cannot read properties of undefined (reading 'lintFiles')

  at lintFiles (node_modules/.pnpm/vite-plugin-eslint2@4.4.0_@types+eslint@8.56.6_eslint@8.57.0_rollup@4.13.0_vite@5.1.6/node_modules/vite-plugin-eslint2/dist/shared/vite-plugin-eslint2.7280f04e.mjs:232:119)
  at MessagePort.<anonymous> (node_modules/.pnpm/vite-plugin-eslint2@4.4.0_@types+eslint@8.56.6_eslint@8.57.0_rollup@4.13.0_vite@5.1.6/node_modules/vite-plugin-eslint2/dist/worker.mjs:38:3)
```

**Comment by GaborTorma** [comment]
You can test with this repo: https://github.com/GaborTorma/nuxt-eslint-vs-code-js-debug-terminal-error2
The error messages shows only when the nuxt.config.js aren't same folder with eslint.config.js like nuxt module development playground.

You need to clone it and start with VS Code and run `pnpm i`
If you run `pnpm dev` in bash -> no error
If you run `pnpm dev` in VS Code JS Debugger terminal -> got error.

I found the problem. When we use in JS Debugger terminal the eslint path is `'eslint'`, in bash `'eslint/use-at-your-own-risk'`.
With `'eslint/use-at-your-own-risk'` works well.

**Comment by GaborTorma** [comment]
workaround with nuxt.config.ts: 

```js
export default defineNuxtConfig({
  modules: [
    '@nuxt/eslint',
  ],

  eslint: {
    checker: {
      eslintPath: 'eslint/use-at-your-own-risk'
    },
  },
})
```

**Comment by antfu** [comment]
Ah I see, we detect flat config under rootDir: https://github.com/nuxt/eslint/blob/7398e4e637cd93e811273ae04ac38077ac72c32f/packages/module/src/modules/checker.ts#L36-L37

I guess, in a way, you have to use that workaround if your config is outside of your root. Or you may do

```ts
  eslint: {
    checker: {
      configType: 'flat'
    },
  },
```

**Comment by antfu** [comment]
When ESLint v9.0 comes up I think we should then make the default to flat so this won't be an issue later on.

**Comment by antfu** [comment]
d84af6fe28bd267de60036417efa1aa9e2495ac0 should fixes it.

**Comment by Bredansky** [comment]
For some reason, I am still experiencing this error. It doesn't affect anything, but looks annoying :)

`nuxt.config.ts`:
```
  eslint: {
    config: {
      stylistic: true,
    },
    checker: true,
  },
  ```
  
  `package.json`:
  ```
    "@nuxt/eslint": "^0.3.13",
    "eslint-plugin-tailwindcss": "^3.17.4",
    "vite-plugin-eslint2": "^4.4.0",
    "nuxt": "^3.12.1",
    ```

`eslint.config.mjs`:
```
import tailwind from 'eslint-plugin-tailwindcss'
import withNuxt from './.nuxt/eslint.config.mjs'

export default withNuxt(
  ...tailwind.configs['flat/recommended'],
  {
    rules: {
      'no-console': ['error', { allow: ['warn', 'error', 'info'] }],
    },
  },
).prepend({
  ignores: ['types'],
})

```

**Comment by GaborTorma** [comment]
This issue is still here if not set the configType to flat.

**Comment by AlvesJorge** [comment]
Can't get this to work either. 
The only way for the nuxt configs to start working is if I open the .nuxt/eslint.config.mjs  file at least once per vscode session
